From 2d0979fb54e025e904c7372666fffbf5dae40f66 Mon Sep 17 00:00:00 2001
From: libplacebo upstream
Subject: [PATCH] renderer: use pass->image for max_peak in hdr_update_peak()

Backport of upstream commit 2d0979fb54e025e904c7372666fffbf5dae40f66.

diff --git a/src/renderer.c b/src/renderer.c
index 0801396..0335e69 100644
--- a/src/renderer.c
+++ b/src/renderer.c
@@ -1184,7 +1184,10 @@ static void hdr_update_peak(struct pass_state *pass)
 {
     const struct pl_render_params *params = pass->params;
     pl_renderer rr = pass->rr;
-    if (!params->peak_detect_params || !pl_color_space_is_hdr(&pass->img.color))
+
+    // Use pass->image than pass->img to deal with cases that input is HDR
+    // content but already linearized when scaling.
+    if (!params->peak_detect_params || !pl_color_space_is_hdr(&pass->image.color))
         goto cleanup;

     if (rr->errors & PL_RENDER_ERR_PEAK_DETECT)
@@ -1196,9 +1199,9 @@ static void hdr_update_peak(struct pass_state *pass)
     if (!rr->gpu->limits.max_ssbo_size)
         goto cleanup;

-    float max_peak = pl_color_transfer_nominal_peak(pass->img.color.transfer) *
+    float max_peak = pl_color_transfer_nominal_peak(pass->image.color.transfer) *
                      PL_COLOR_SDR_WHITE;
-    if (pass->img.color.transfer == PL_COLOR_TRC_HLG)
+    if (pass->image.color.transfer == PL_COLOR_TRC_HLG)
         max_peak = pass->img.color.hdr.max_luma;
     if (max_peak <= pass->target.color.hdr.max_luma + 1e-6)
         goto cleanup; // no adaptation needed
--
2.51.0.windows.1
